src/components/ProtectedRoute/ProtectedRoute.jsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+
+const ProtectedRoute = ({ children, ...props }) => {
+  const gotToken = localStorage.getItem('token');
+
+  return (
+    <Route
+      {...props}
+      render={() => (gotToken ? children : <Redirect to="/" />)}
```

Comment:
Nit: 
How about just `token` instead of `gotToken`? In my mind, the fact that it exists implies that it was gotten.

As I said earlier today, one of my favorite software engineering quotes is:

"There are only two hard things in Computer Science: cache invalidation and naming things." -- Phil Karlton